src/main/kotlin/no/nav/amt/aktivitetskort/service/AktivitetskortService.kt

```diff
@@ -74,21 +73,16 @@ class AktivitetskortService(
 		}
 	}
 
-	private fun getAktivitetskortId(deltakerId: UUID, tiltakstype: Tiltak.Type): UUID {
-		val aktivitetskortId = amtArenaAclClient.getArenaIdForAmtId(deltakerId)
-			?.let { aktivitetArenaAclClient.getAktivitetIdForArenaId(it) }
+	private fun getAktivitetskortId(deltakerId: UUID): UUID {
+		val eksisterendeAktivitetskortId = aktivitetskortIdForDeltaker(deltakerId)
 
-		if (aktivitetskortId != null) {
-			return aktivitetskortId
-		} else if (kometErMasterForTiltakstype(tiltakstype) || EnvUtils.isDev()) {
-			return aktivitetskortIdForDeltaker(deltakerId)
-		} else {
-			throw IllegalStateException("Kunne ikke hente aktivitetskortId for deltaker med id $deltakerId")
-		}
+		return eksisterendeAktivitetskortId
+			?: amtArenaAclClient.getArenaIdForAmtId(deltakerId)
+				?.let { aktivitetArenaAclClient.getAktivitetIdForArenaId(it) }
+			?: UUID.randomUUID().also { log.info("Definerer egen aktivitetskortId: $it for deltaker med id $deltakerId") }
```

Comment:
Grunnen til at vi hadde toggle på om man får lov til å opprette en ny id for aktivitetskort var et forsøk på å beskytte mot at vi oppretter ny kort bare fordi f.eks. arena-acl eller appen til DAB er nede litt. Men siden vi sjekker om vi allerede har sendt et aktivitetskort tidligere og evt bruker id-en derfra kan det kanskje gå bra. Men det kommer an på om DAB leser tiltaksdeltakelser fra arena og oppretter noe for dem, eller om de kun får aktivitetskort for tiltaksdeltakelser på våre typer fra oss. Hvis de også lytter på arena sine tiltaksdeltakelser så kan vi fort få doble kort hvis vi fjerner togglen.

Reply:
Det ble flere scenarioer her
1. Gitt at vi har opprettet deltakelsen så kan vi generere egen id og dab kaster alle disse med EKSTERN_ID fra arena så her trenger vi ikke tenke på rekkefølge, de har da ingen oversikt over aktivitetskortID=>arenaId
2. Gitt at det er en tiltakstype som vi "eier"(ikke AFT), så må vi spørre dab om en id slik at de kan lage mapping mellom arenaid og aktivitetskortId. Om de da leser deltakeren etter at vi har spurt om id så kaster de arena sine senere endringer

Når det gjelder midlertidig nedetid så ser jeg at amt-arena-acl klienten kaster exception hvis man får en annen feil enn 404 og aktivitetskort-acl klientene kaster exception ved alle typer feil så ser ut som det er håndtert